tests/test_spark_users.py

```diff
@@ -0,0 +1,137 @@
+    }
+
+    try:
+        utils.upload_dcos_test_jar()
+        utils.require_spark(service_name=SERVICE_NAME, additional_options=options, zk=DISPATCHER_ZK)
+        yield
+    finally:
+        utils.teardown_spark(service_name=SERVICE_NAME, zk=DISPATCHER_ZK)
+
+
+@pytest.mark.sanity
+@pytest.mark.parametrize('user,use_ucr_containerizer,use_ucr_for_spark_submit', [
+    ("nobody", True, True),
+    ("nobody", True, False),
+    ("nobody", False, True),
+    ("nobody", False, False),
+    ("root", True, True),
+    ("root", True, False),
+    ("root", False, True),
+    ("root", False, False)
+])
+def test_user_propagation(setup_spark,  user, use_ucr_containerizer, use_ucr_for_spark_submit):
+    dispatcher_task = utils.get_dispatcher_task(service_name=SERVICE_NAME)
+    _check_task_user(dispatcher_task, user, use_ucr_containerizer)
+
+    extra_args = []
+    if use_ucr_for_spark_submit:
+        extra_args = ["--conf spark.mesos.containerizer=mesos"]
+
+    _submit_job_and_verify_users(user, use_ucr_for_spark_submit, extra_args=extra_args)
+
+
+@pytest.mark.sanity
+@pytest.mark.parametrize('user,user_override,use_ucr_containerizer,use_ucr_for_spark_submit', [
+    ("root", "nobody", True, True),
+    ("root", "nobody", True, False),
+    ("root", "nobody", False, True),
+    ("root", "nobody", False, False)
+])
+def test_user_overrides(setup_spark,  user, user_override, use_ucr_containerizer, use_ucr_for_spark_submit):
+    """This test uses overrides for 'root' user only to lower down total amount of tests
+       while preserving containerizers combinations. User override functionality is independent
+       from any specific username, thus submitting jobs to Dispatcher running as 'root' with
+       jobs running as 'nobody' is sufficient to test it.
+    """
+    dispatcher_task = utils.get_dispatcher_task(service_name=SERVICE_NAME)
+    _check_task_user(dispatcher_task, user, use_ucr_containerizer)
+
+    extra_args = [f"--conf spark.mesos.driverEnv.SPARK_USER={user_override}"]
+    if use_ucr_for_spark_submit:
+        extra_args = extra_args + ["--conf spark.mesos.containerizer=mesos"]
+    else:
+        extra_args = extra_args + [f"--conf spark.mesos.executor.docker.parameters=user={user_override}"]
+
+    _submit_job_and_verify_users(user_override, use_ucr_for_spark_submit, extra_args=extra_args)
+
+
+def _submit_job_and_verify_users(user, use_ucr_for_spark_submit, extra_args=[]):
+    app_name = "MockTaskRunner"
+
+    submit_args = ["--conf spark.cores.max=1",
+                   "--class {}".format(app_name)] + extra_args
+
+    driver_task_id = utils.submit_job(service_name=SERVICE_NAME,
+                                      app_url=utils.dcos_test_jar_url(),
+                                      app_args="1 300",
+                                      args=submit_args)
+    try:
+        sdk_tasks.check_running(app_name, 1, timeout_seconds=300)
+        driver_task = shakedown.get_task(driver_task_id, completed=False)
```

Comment:
we ought to move away from `shakedown` but it doesn't have to be this PR

Reply:
I created a Jira ticket for it, probably it makes sense to wait until we have proper dependency management for the testing package from [dcos-commons](https://github.com/mesosphere/dcos-commons).